tests/run-pass/fs.rs

```diff
@@ -19,10 +25,23 @@ fn main() {
     test_errors();
     test_rename();
     test_directory();
+    test_dup_stdout_stderr();
 }
 
 fn tmp() -> PathBuf {
-    std::env::var("MIRI_TEMP").map(PathBuf::from).unwrap_or_else(|_| std::env::temp_dir())
+    std::env::var("MIRI_TEMP")
+        .map(|tmp| {
+            // MIRI_TEMP is set outside of our emulated
+            // program, so it may have path separators that don't
+            // correspond to our target platform. We normalize them here
+            // before constructing a `PathBuf`
+
+            #[cfg(windows)]
+            return PathBuf::from(tmp.replace("/", "\\"));
+
+            #[cfg(not(windows))]
+            return PathBuf::from(tmp.replace("\\", "/"));
+        }).unwrap_or_else(|_| std::env::temp_dir())
```

Comment:
I start to wonder if we should put this in a lazy_static to avoid recomputing all the time.^^ But that is a question for another PR.